inc/sitemaps/class-post-type-sitemap-provider.php

```diff
@@ -626,7 +656,9 @@ protected function get_url( $post ) {
 		 *
 		 * @see https://wordpress.org/plugins/page-links-to/ can rewrite permalinks to external URLs.
 		 */
-		if ( $this->get_classifier()->classify( $url['loc'] ) === WPSEO_Link::TYPE_EXTERNAL ) {
+		$post_type_has_front = $this->get_post_type_has_front();
+
+		if ( $this->get_classifier()->classify( $url['loc'] ) === WPSEO_Link::TYPE_EXTERNAL && $post_type_has_front ) {
```

Comment:
I'd like to we don't touch this code. It's possible to pass _home_url_ without path (if with_front is true) in  instantiation  of_ _WPSEO_Link_Type_Classifier_ (it runs only once).

Reply:
The problem is that _home_url_ and _site_url_ both contain the path, see [#9634](https://github.com/Yoast/wordpress-seo/issues/9634). For with_front set to true this is fine, but with false it is not.

I've updated it to remove the path from _home_url_ if the post type with_front rewrite option is set to false.